pyirf/io/gadf.py

```diff
@@ -325,3 +309,177 @@ def create_rad_max_hdu(
+
+    for file_name in files[1:]:
+        with fits.open(file_name) as hdul:
+            data = hdul['THETA_CUTS'].data
+            if (data != data0).any():
+                print("difference between file: " + files[0] + " and " + file_name + " in cut values: " + extname)
+                return False
+
+    return True
+
+
+def read_fits_bins_lo_hi(file_name, extname, tag):
+    """
+    Reads from a fits file two arrays of tag_LO and tag_HI.
+    If more then one file is given it checks that all of them are consistent before returning the value
+
+    Parameters
+    ----------
+    file_name: string or list of string
+        file to be read, if a list of
+    extname: string
+        name of the extension to read the data from in fits file
+    tag: string
+        name of the field in the extension to extract, _LO and _HI will be added
+
+    Returns
+    -------
+    bins: astropy.units.Quantity[energy]
+        bins
+    """
+
+    tag_lo = tag + '_LO'
+    tag_hi = tag + '_HI'
+
+    # to allow the function work on either single file or a list of files convert a single string into a list
+    if isinstance(file_name, str):
+        file_name = [file_name]
+
+    old_table = None
+    for this_file in file_name:
+        table = QTable.read(this_file, hdu=extname)
+        if old_table is not None:
+            if not old_table[tag_lo][0].shape == table[tag_lo][0].shape:
+                raise ValueError('Non matching bins')
+            if not ((old_table[tag_lo][0] == table[tag_lo][0]).all() and (old_table[tag_hi][0] == table[tag_hi][0]).all()):
+                raise ValueError('Non matching bins')
+        old_table = table
+
+    return table[tag_lo][0], table[tag_hi][0]
+
+
+def read_irf_grid(files, extname, field_name):
+    """
+    Reads in a grid of IRFs for a bunch of different parameters and stores them in lists
+
+    Parameters
+    ----------
+    files: string or list of strings
+        files to be read
+    extname: string
+        name of the extension to read the data from in fits file
+    field_name: string
+        name of the field in the extension to extract
+
+    Returns
+    -------
+    irfs_all: np.array
+        array of IRFs, first axis specify the file number(if more then one is given),
+        second axis is the offset angle, rest of the axes are IRF-specific
+    theta_bins: astropy.units.Quantity[angle]
+        theta bins
+    """
+
+    # to allow the function work on either single file or a list of files convert a single string into a list
+    if isinstance(files, str):
+        files = [files]
+
+    n_files = len(files)
+
+    irfs_all = np.empty(n_files, dtype=np.object)
+    for ifile, this_file in enumerate(files):
+        # [0] because there the IRFs are written as a single row of the table
+        irfs_all[ifile] = QTable.read(this_file, hdu=extname)[field_name][0]
+
+    # if the function is run on single file do not need the first axis dimension
+    if n_files == 1:
```

Comment:
@maxnoe  I'm not sure if I understand correctly this comment. 
do you mean to return the boolean together with the IRFs ?
also in the first paragraph you mentioned that the return type/shape should be always conserved
so I can change the code to return also a list of one IRF if only one argument is passed, but then I think this boolean would not be useful, because one could just check the dimention of the returned array,
or would returning two different possible shapes depending on this returned boolean be fine with you?